optimize/backend/src/main/java/io/camunda/optimize/license/CamundaLicense.java

```diff
@@ -0,0 +1,112 @@
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.license;
+
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
+import org.camunda.bpm.licensecheck.LicenseKeyImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CamundaLicense {
+
+  public static final String CAMUNDA_LICENSE_ENV_VAR_KEY = "CAMUNDA_LICENSE_KEY";
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamundaLicense.class);
+  private static final String UNKNOWN_LICENSE_TYPE = "unknown";
+  private boolean isValid;
+  private LicenseType licenseType;
+  private boolean isInitialized;
+
+  // @VisibleForTesting
+  protected CamundaLicense() {}
+
+  public CamundaLicense(final String license) {
+    initializeWithLicense(license);
+  }
+
+  public synchronized boolean isValid() {
+    return isValid;
+  }
+
+  public synchronized LicenseType getLicenseType() {
+    return licenseType;
+  }
+
+  public synchronized void initializeWithLicense(final String license) {
+    if (isInitialized) {
+      return;
+    }
+
+    if (license != null && !license.isBlank()) {
+      isValid = determineLicenseValidity(license);
+      licenseType = getLicenseTypeFromProperty(license);
+    } else {
+      isValid = false;
+      licenseType = LicenseType.UNKNOWN;
+      LOGGER.error(
+          "No license detected when one is expected. Please provide a license through the "
+              + CAMUNDA_LICENSE_ENV_VAR_KEY
+              + " environment variable.");
+    }
+
+    isInitialized = true;
+  }
+
+  /**
+   * Camunda licenses will have a license type property, fetch that out of the license and return
+   * the value
+   *
+   * <p>Self-managed mode is any other possibility. (ex, blank license, prop missing, etc)
+   */
+  private LicenseType getLicenseTypeFromProperty(final String licenseStr) {
+    try {
+      final LicenseKey licenseKey = getLicenseKey(licenseStr);
+      final String licenseType =
+          licenseKey.getProperties().getOrDefault("licenseType", UNKNOWN_LICENSE_TYPE);
+
+      if (UNKNOWN_LICENSE_TYPE.equals(licenseType)) {
+        LOGGER.error(
+            "Expected a licenseType property on the Camunda License, but none were found.");
+      }
+
+      return LicenseType.get(licenseType);
+    } catch (final InvalidLicenseException e) {
+      LOGGER.error(
+          "Expected a valid license when determining the type of license, but encountered an invalid one instead. ",
+          e);
+    } catch (final Exception e) {
+      LOGGER.error(
+          "Expected to determine the license type of the license, but the following unexpected error was encountered: ",
+          e);
+    }
+    return LicenseType.UNKNOWN;
```

Comment:
❓ we still just return the unknown type and would validate with`initializeWithLicense` as long as the license itself was valid. Should this be blocking validation or is a valid license with unknown type still fine (or maybe it would never be valid in that case)?